embed_grader/serapis/views/courses.py

```diff
@@ -169,6 +171,88 @@ def unenroll_course(request, course_id):
     }
     return render(request, 'serapis/unenroll_course.html', template_context)
 
+@login_required(login_url='/login/')
+def download_csv(request, course_id):
+    user = User.objects.get(username=request.user)
+    course = Course.objects.get(id=course_id)
+    print(course)
+    students = []
+    headers = ['Last Name', 'First Name', 'UID']
+
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = 'attachment; filename="%s_scores.csv"' % course
+    writer = csv.writer(response)
+
+    if not user.has_perm('download_csv', course):
+        return HttpResponse("Not enough privilege.")
+
+    assignment_list = Assignment.objects.filter(course_fk=course_id).order_by('-id')
+    for h in assignment_list:
+        headers.append(h.name)
+
+    writer.writerow(headers)
+
+    assignment_ids = []
+
+    for assign in assignment_list:
+        assignment_ids.append(assign.id)
+
+    print(assignment_ids)
+
+    students = [o.user_fk for o in CourseUserList.objects.filter(course_fk=course)]
+    scores = {}
+    for s in students:
+        scores[s] = [0. for _ in range(len(assignment_list))]
+    for idx, aid in enumerate(assignment_ids):
+        assignment = Assignment.objects.get(id=aid)
+        teams = Team.objects.filter(assignment_fk=assignment)
+        for team in teams:
+            last_submission = grading.get_last_submission(team, assignment)
+        if last_submission is None:
+            continue
+        _, score, _ = last_submission.retrieve_task_grading_status_and_score_sum(True)
+        team_students = [o.user_fk for o in TeamMember.objects.filter(team_fk=team)]
+        for s in team_students:
+            if s.id != 1:
+                scores[s][idx] = score
+
+    for s in scores:
+        new_row = []
+        user_profile = UserProfile.objects.get(user=s)
+        terms = [user_profile.uid] + list(map(str, scores[s]))
+        new_row.append(user_profile.user.last_name)
+        new_row.append(user_profile.user.first_name)
+        new_row.append(user_profile.uid)
+        count = 0
+        sum_scores_for_each_assignment = [None]*(len(terms)-1)
+        max_score_for_each_assignment = [None]*(len(terms)-1)
+        for t in terms[1:]:
+            if sum_scores_for_each_assignment[count] == None:
```

Comment:
Let's initialize `sum_scores_for_each_assignment` as `0.` so that you don't need to have this branch?